ion-schema/src/constraint.rs

```diff
@@ -252,6 +253,22 @@ impl Constraint {
         Constraint::Fields(FieldsConstraint::new(fields, true))
     }
 
+    /// Creates a [Constraint::FieldNames] referring to the type represented by the provided [TypeId] where each field name is distinct.
+    pub fn distinct_field_names(type_id: TypeId) -> Constraint {
+        Constraint::FieldNames(FieldNamesConstraint::new(
+            TypeReference::new(type_id, NullabilityModifier::Nothing),
+            true,
+        ))
+    }
+
+    /// Creates a [Constraint::FieldNames] referring to the type represented by the provided [TypeId] where each field name.
+    pub fn field_names(type_id: TypeId) -> Constraint {
+        Constraint::FieldNames(FieldNamesConstraint::new(
+            TypeReference::new(type_id, NullabilityModifier::Nothing),
+            false,
+        ))
+    }
```

Comment:
No particular reason since these methods are all part of a module which is not public we can choose either way so probably a single method will be better.